espresso/environment/6_batch_inbox_test.go

```diff
@@ -0,0 +1,169 @@
+// This way we trick batcher into posting unauthenticated batches to batch inbox.
+// We then verify that these batches aren't accepted by the batch inbox contract and derivation pipeline.
+//
+// The test is defined as follows
+// Arrange:
+//
+//	Deploy a mock BatchAuthenticator.
+//	Configure batcher to use said authenticator instead of the real one.
+//	Start sequencer, batcher in Espresso mode and OP node.
+//
+// Assert:
+//
+//	Assert that transaction submitting the batch was reverted by
+//	batch inbox contract
+//	Assert that derivation pipeline doesn't progress
+func TestE2eDevNetWithoutAuthenticatingBatches(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+
+	system, _, err :=
+		launcher.StartDevNet(ctx, t,
+			env.Config(func(cfg *e2esys.SystemConfig) {
+				cfg.DisableBatcher = true
+			}),
+		)
+
+	if have, want := err, error(nil); have != want {
+		t.Fatalf("failed to start dev environment with espresso dev node:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+	}
+
+	batchDriver := system.BatchSubmitter.TestDriver()
+	// Set mock batcher authenticator address
+	batchDriver.BatchSubmitter.RollupConfig.BatchAuthenticatorAddress = common.Address{}
+
+	// Substitute batcher's transaction manager with one that always sends transactions, even
+	// if they won't succeed. Otherwise batcher wouldn't submit transactions that would revert to
+	// batch inbox
+	txMgrCliConfig := setuputils.NewTxMgrConfig(system.NodeEndpoint(e2esys.RoleL1), system.Cfg.Secrets.Batcher)
+	txMgrConfig, err := txmgr.NewConfig(txMgrCliConfig, log.Root())
+	require.NoError(t, err)
+	txMgrConfig.Backend = AlwaysSendingETHBackend{
+		inner: txMgrConfig.Backend,
+	}
+	txMgr, err := txmgr.NewSimpleTxManagerFromConfig("always-sending", log.Root(), &metrics.NoopTxMetrics{}, txMgrConfig)
+	require.NoError(t, err)
+	batchDriver.Txmgr = txMgr
+
+	// Start the batcher
+	err = batchDriver.StartBatchSubmitting()
+	l1Client := system.NodeClient(e2esys.RoleL1)
+
+	// Wait for batcher to submit a transaction to BatchInbox
+	var batchInboxTxHash common.Hash
+	for {
+		l1Height, err := l1Client.BlockNumber(ctx)
+		require.NoError(t, err)
+		_, err = geth.FindBlock(l1Client,
+			0,
+			int(l1Height),
+			time.Minute*2,
+			func(block *types.Block) (bool, error) {
+				for _, tx := range block.Transactions() {
+					if *tx.To() == system.RollupConfig.BatchInboxAddress {
+						batchInboxTxHash = tx.Hash()
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+		if err == nil {
+			break
+		}
+	}
+
+	receipt, err := l1Client.TransactionReceipt(ctx, batchInboxTxHash)
+	require.NoError(t, err)
+
+	require.Equal(t, receipt.Status, types.ReceiptStatusFailed, "transaction should've been rejected by BatchInbox contract")
+
+	_, err = geth.WaitForBlockToBeSafe(new(big.Int).SetUint64(1), system.NodeClient(e2esys.RoleVerif), time.Minute)
+	require.Error(t, err)
```

Comment:
Would it be helpful to also assert that `SafeL2` remains the same by querying `SyncStatus`? In this way, we can verify the state directly besides relying on the timeout.

Reply:
We'd still need to wait some amount of time to distinguish between pipeline not _yet_ approving SafeL2 and pipeline stalling